packages/@glimmer/runtime/test/debugger-test.ts

```diff
@@ -1,82 +1,94 @@
-import { UpdatableReference } from "@glimmer/object-reference";
-import {
-  IteratorResult,
-  RenderResult,
-  resetDebuggerCallback,
-  setDebuggerCallback,
-  Template,
-} from "@glimmer/runtime";
-import {
-  TestDynamicScope,
-  TestEnvironment,
-} from "@glimmer/test-helpers";
-import { Opaque } from '@glimmer/util';
+import { RenderTests, module, test } from '@glimmer/test-helpers';
+import { setDebuggerCallback } from '@glimmer/runtime';
 
-let env: TestEnvironment;
-let root: HTMLElement;
-let result: RenderResult;
-let self: UpdatableReference<Opaque>;
+class DebuggerTests extends RenderTests {
+  @test "basic debugger statement"() {
+    let expectedContext = {
+      foo: 'bar',
+      a: {
+        b: true
+      }
+    };
+    let callbackExecuted = 0;
 
-function compile(template: string) {
-  return env.compile(template);
-}
+    setDebuggerCallback((context: any, get) => {
+      callbackExecuted++;
+      this.assert.equal(context.foo, expectedContext.foo);
+      this.assert.equal(get('foo'), expectedContext.foo);
+    });
 
-function commonSetup() {
-  env = new TestEnvironment(); // TODO: Support SimpleDOM
-  root = document.createElement('div');
-}
+    this.render('{{#if a.b}}true{{debugger}}{{else}}false{{debugger}}{{/if}}', expectedContext);
+    this.assert.equal(callbackExecuted, 1);
+    this.assertHTML('true');
 
-function render<T>(template: Template<T>, context={}) {
-  self = new UpdatableReference(context);
-  env.begin();
-  let templateIterator = template.render({ self, parentNode: root, dynamicScope: new TestDynamicScope() });
-  let iteratorResult: IteratorResult<RenderResult>;
-  do {
-    iteratorResult = templateIterator.next();
-  } while (!iteratorResult.done);
+    this.assertStableRerender();
+    this.assert.equal(callbackExecuted, 1);
+    this.assertHTML('true');
 
-  result = iteratorResult.value;
-  env.commit();
-  return result;
-}
+    expectedContext = {
+      foo: 'baz',
+      a: {
+        b: false
+      }
+    };
+    this.rerender(expectedContext);
+    this.assert.equal(callbackExecuted, 2);
+    this.assertHTML('false');
```

Comment:
Had failing specs as `this.snapshot` is outdated in `assertStableNodes()` after the rerender
https://github.com/glimmerjs/glimmer-vm/blob/aa0ddc6ee44083a14aa690f7f212e7f67127ab1c/packages/%40glimmer/test-helpers/lib/abstract-test-case.ts#L522-L526

I resorted to calling `assertStableRerender` to have a new snapshot taken but feel that it isn't the right thing to do. Would appreciate some input regarding this @chadhietala 🙇